firmware/src/widgets/stockwidget/StockWidget.cpp

```diff
@@ -46,42 +46,28 @@ void StockWidget::draw(bool force) {
 
 void StockWidget::update(bool force) {
     if (force || m_stockDelayPrev == 0 || (millis() - m_stockDelayPrev) >= m_stockDelay) {
-        if (m_taskHandle == NULL) { // Check if task is not already running
-            setBusy(true);
-            // Create a task to handle all stock updates
-            if (xTaskCreate(taskGetStockData, "StockDataTask", 8192, this, 1, &m_taskHandle) == pdPASS) {
-                Serial.println("StockDataTask created");
-                m_stockDelayPrev = millis();
-            } else {
-                Serial.println("Failed to create StockDataTask");
-                setBusy(false);
-            }
-        } else {
-            // Serial.println("StockDataTask is already running");
+        setBusy(true);
```

Comment:
each individual orb calling `setBusy` does not make sense, especially when it is no longer blocking.
Instead, the request wrapper should do this, calling `setBusy(true)` when it starts handling a request, and `setBusy(false)` when the request is handled (or something along those lines)

---

The whole thing where there are multiple code paths trying to control one resouce (a led) already did not make sense before, but now it is even worse imo.

Reply:
yeah I agree

Reply:
I created a Utils::setBusy()